provider/lxd/export_test.go

```diff
@@ -12,6 +14,32 @@ var (
 	NewInstance = newInstance
 )
 
+func NewProviderWithMocks(
```

Comment:
Since these types are public (they don't need exporting to the test package) I would have them as methods on the suite.

Reply:
but `environProvider` isn't public and we need to use `export_test.go` to bridge the gap between the `lxd` package and `lxd_test` package, otherwise I can't reach them. 

https://medium.com/@robiplus/golang-trick-export-for-test-aa16cbd7b8cd